app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -85,6 +89,50 @@ public static boolean isTv(final Context context) {
         return DeviceUtils.isTV;
     }
 
+    public static boolean isDesktopMode(final Context context) {
+        if (ContextCompat.getSystemService(context, UiModeManager.class)
+                .getCurrentModeType() == Configuration.UI_MODE_TYPE_DESK) {
+            return true;
+        }
+        // DeX check for standalone and multi-window mode
+        try {
+            final Configuration config = context.getResources().getConfiguration();
+            final Class<?> configClass = config.getClass();
+            final int semDesktopModeEnabledConst =
+                    configClass.getField("SEM_DESKTOP_MODE_ENABLED").getInt(configClass);
+            final int currentMode =
+                    configClass.getField("semDesktopModeEnabled").getInt(config);
+            if (semDesktopModeEnabledConst == currentMode) {
+                return true;
+            }
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            // empty
+        }
+        @SuppressLint("WrongConstant") final Object desktopModeManager = context
+                .getApplicationContext()
+                .getSystemService("desktopmode");
+        if (desktopModeManager != null) {
+            try {
+                final Method getDesktopModeStateMethod = desktopModeManager.getClass()
+                        .getDeclaredMethod("getDesktopModeState");
+                final Object desktopModeState = getDesktopModeStateMethod
+                        .invoke(desktopModeManager);
+                final Class<?> desktopModeStateClass = desktopModeState.getClass();
+                final Method getEnabledMethod = desktopModeStateClass
+                        .getDeclaredMethod("getEnabled");
+                final int enabled = (int) getEnabledMethod.invoke(desktopModeState);
+                if (enabled == desktopModeStateClass
+                        .getDeclaredField("ENABLED").getInt(desktopModeStateClass)) {
+                    return true;
+                }
+            } catch (NoSuchFieldException | NoSuchMethodException
+                    | IllegalAccessException | InvocationTargetException e) {
+                // Device does not support DeX 3.0
+            }
+        }
+        return false;
+    }
+
```

Comment:
Note: 
This always executed on ALL devices when the VideoDetailFragment is initialized. 
Combined with the use of Reflection - which is also not that performant (https://stackoverflow.com/a/435568) - this might be an bigger issue.

Reply:
How long do you think  a one-time check will take? On an old device this code takes <0.001ms and is run only once in view init. Line 114 guards against any reflection calls and on line 101 there is one reflection call which will throw and fail fast on non-desktop devices.

Reply:
@litetex performance is negligible here. This is a one-time check, and even though reflection is ten times slower than normal class access, according to the SO you linked it still takes <10^-6s, which is not a problem. Maybe let's just add a note in the javadoc so that this method is not misused in the future.

Reply:
Yup, I have added a jsdoc comment with a note that this should ideally only be invoked once on view init